mgmtsystem/scvmm.py

```diff
@@ -275,31 +275,34 @@ def does_template_exist(self, template):
         return len(result) > 0
 
     def deploy_template(self, template, host_group, vm_name=None, **kwargs):
-        timeout = kwargs.pop('timeout', 900)
-        vm_cpu = kwargs.get('cpu', 0)
-        vm_ram = kwargs.get('ram', 0)
-        script = """
-            $tpl = Get-SCVMTemplate -Name "{template}" -VMMServer $scvmm_server
-            $vm_hg = Get-SCVMHostGroup -Name "{host_group}" -VMMServer $scvmm_server
-            $vmc = New-SCVMConfiguration -VMTemplate $tpl -Name "{vm_name}" -VMHostGroup $vm_hg
-            Update-SCVMConfiguration -VMConfiguration $vmc
-            New-SCVirtualMachine -Name "{vm_name}" -VMConfiguration $vmc
-        """.format(
-            template=template,
-            vm_name=vm_name,
-            host_group=host_group)
-        if vm_cpu:
-            script += " -CPUCount '{vm_cpu}'".format(vm_cpu=vm_cpu)
-        if vm_ram:
-            script += " -MemoryMB '{vm_ram}'".format(vm_ram=vm_ram)
-        self.logger.info(" Deploying SCVMM VM `{}` from template `{}` on host group `{}`"
-            .format(vm_name, template, host_group))
-        self.run_script(script)
-        self.enable_virtual_services(vm_name)
-        self.start_vm(vm_name)
-        self.wait_vm_running(vm_name, num_sec=timeout)
-        self.update_scvmm_virtualmachine(vm_name)
-        return vm_name
+        if self.does_template_exist(template):
+            timeout = kwargs.pop('timeout', 900)
+            vm_cpu = kwargs.get('cpu', 0)
+            vm_ram = kwargs.get('ram', 0)
+            script = """
+                $tpl = Get-SCVMTemplate -Name "{template}" -VMMServer $scvmm_server
+                $vm_hg = Get-SCVMHostGroup -Name "{host_group}" -VMMServer $scvmm_server
+                $vmc = New-SCVMConfiguration -VMTemplate $tpl -Name "{vm_name}" -VMHostGroup $vm_hg
+                Update-SCVMConfiguration -VMConfiguration $vmc
+                New-SCVirtualMachine -Name "{vm_name}" -VMConfiguration $vmc
+            """.format(
+                template=template,
+                vm_name=vm_name,
+                host_group=host_group)
+            if vm_cpu:
+                script += " -CPUCount '{vm_cpu}'".format(vm_cpu=vm_cpu)
+            if vm_ram:
+                script += " -MemoryMB '{vm_ram}'".format(vm_ram=vm_ram)
+            self.logger.info(" Deploying SCVMM VM `{}` from template `{}` on host group `{}`"
+                .format(vm_name, template, host_group))
+            self.run_script(script)
+            self.enable_virtual_services(vm_name)
+            self.start_vm(vm_name)
+            self.wait_vm_running(vm_name, num_sec=timeout)
+            self.update_scvmm_virtualmachine(vm_name)
+            return vm_name
+        else:
+            self.logger.warn("Template {} does not exist".format(template))
```

Comment:
I added the exception and ran a negative test to force it to fail.  Here's the output.
        else:
            self.logger.warn("Template {} does not exist".format(template))
>           raise self.PowerShellScriptError("Template {} does not exist".format(template))
E           PowerShellScriptError: Template fedora-tmpl-sm.vhd does not exist
../lib/python2.7/site-packages/mgmtsystem/scvmm.py:306: PowerShellScriptError
================================================================================= 1 failed in 48.64 seconds ==================================================================================
collecting artifacts